netbox/dcim/models/racks.py

```diff
@@ -235,13 +240,26 @@ def clean(self):
                 position__isnull=True
             ).order_by('-position').first()
             if top_device:
-                min_height = top_device.position + top_device.device_type.u_height - 1
+                min_height = top_device.position + top_device.device_type.u_height - self.starting_unit
                 if self.u_height < min_height:
                     raise ValidationError({
                         'u_height': "Rack must be at least {}U tall to house currently installed devices.".format(
                             min_height
                         )
                     })
+            # Validate the Rack starting_unit > position of any installed devices
+            first_device = Device.objects.filter(
+                rack=self
+            ).exclude(
+                position__isnull=True
+            ).order_by('position').first()
```

Comment:
We should be able to reuse the above queryset and just call `first()` and `last()` to get the top and bottom devices, respectively.